interfaces/builtin/network_setup_control.go

```diff
@@ -19,17 +19,31 @@
 
 package builtin
 
+import "github.com/snapcore/snapd/release"
+
 const networkSetupControlConnectedPlugAppArmor = `
 # Description: Can read/write netplan configuration files
 
 /etc/netplan/{,**} rw,
 /etc/network/{,**} rw,
 `
 
+const networkSetupControlConnectedPlugAppArmorClassic = `
+# Description: Can read/write NetworkManager's dispatcher (nm-dispatcher) scripts
+
+/etc/NetworkManager/dispatcher.d/{,**} rw,
+`
+
 func init() {
+
+	var classicAppArmor string
+	if release.OnClassic {
+		classicAppArmor = networkSetupControlConnectedPlugAppArmorClassic
+	}
+
 	registerIface(&commonInterface{
 		name: "network-setup-control",
-		connectedPlugAppArmor: networkSetupControlConnectedPlugAppArmor,
+		connectedPlugAppArmor: networkSetupControlConnectedPlugAppArmor + classicAppArmor,
 		reservedForOS:         true,
 	})
 }
```

Comment:
Please create a test case in interfaces/builtin/network_setup_control_test.go that your rule is added on OnClassic and not added when not OnClassic. See TestConnectedPlugSnippetUsesUnconfinedLabelOnClassic in upower_observe_test.go for inspiration.